File: official/docs/csharp/current/billing/add-stripe-bank-account.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using EasyPost;
using Newtonsoft.Json;

namespace EasyPostExamples
{
    public class Examples
    {
        public static async Task Main()
        {
            var client = new EasyPost.Client(new EasyPost.ClientConfiguration("EASYPOST_API_KEY"));

            EasyPost.Models.API.PaymentMethod paymentMethod = await Client.ReferralCustomer.AddBankAccountFromStripe(
                "REFERRAL_USER_API_KEY",
                "fca_...",
                new Dictionary<string, object>
                {
                    { "ip_address", "127.0.0.1" },
                    { "user_agent", "Mozilla/5.0" },
                    { "accepted_at", 1722510730 }
                },
                EasyPost.Models.API.PaymentMethod.Priority.Primary
            );

            Console.WriteLine(JsonConvert.SerializeObject(paymentMethod, Formatting.Indented));
        }
    }
}

File: official/docs/ruby/current/billing/add-stripe-bank-account.rb
require 'easypost'

client = EasyPost::Client.new(api_key: 'EASYPOST_API_KEY')

payment_method = client.referral_customer.add_bank_account_from_stripe(
  'REFERRAL_USER_API_KEY',
  'fca_...',
  {
    ip_address: '127.0.0.1',
    user_agent: 'Mozilla/5.0',
    accepted_at: 172_251_073,
  },
  'primary',
)

puts payment_method
